src/tabs/__integ__/tabs.test.ts

```diff
@@ -302,3 +315,89 @@ test(
   })
 );
+
+test(
+  'verifies focus moves to dismissible via keyboard',
+  setupTest(async page => {
+    await page.focusTabHeader();
+    await page.click(wrapper.findTabLinkByIndex(3).toSelector());
+    await page.keys(['ArrowRight']);
+    // Since the focus doesn't actually update on the action/dismissible buttons,
+    // we evaluate a valid focus by making it sure it hasn't moved to the next tab.
+    await expect(page.isDisplayed(wrapper.findTabLinkByIndex(3).toSelector())).resolves.toBe(true);
+    await expect(page.isFocused(wrapper.findTabLinkByIndex(4).toSelector())).resolves.toBe(false);
+  })
+);
+
+test(
+  'verifies focus moves from dismissible to tab content on tab press',
+  setupTest(async page => {
+    await page.focusTabHeader();
+    await page.click(wrapper.findTabLinkByIndex(3).toSelector());
+    await page.keys(['ArrowRight']);
+    await page.keys(['Tab']);
+    await expect(page.isFocused(wrapper.findTabContent().toSelector())).resolves.toBe(true);
+  })
+);
+
+test(
+  'verifies focus moves from dismissible button after the dismissible button is clicked',
+  setupTest(async page => {
+    await page.focusTabHeader();
+    await page.click(wrapper.findTabLinkByIndex(3).toSelector());
+    await page.keys(['ArrowRight']);
+    await page.keys(['Tab']);
+    await page.keys(['Enter']);
+    await expect(page.isFocused(wrapper.findTabLinkByIndex(3).toSelector())).resolves.toBe(false);
+  })
+);
+
+test(
+  'verifies focus moves to action via keyboard',
+  setupTest(async page => {
+    await page.focusTabHeader();
+    await page.click(wrapper.findTabLinkByIndex(6).toSelector());
+    await page.keys(['ArrowRight']);
+    await page.keys(['ArrowRight']);
+    // Since the focus doesn't actually update on the action/dismissible buttons,
+    // we evaluate a valid focus by making it sure it hasn't moved to the next tab.
+    await expect(page.isDisplayed(wrapper.findTabLinkByIndex(6).toSelector())).resolves.toBe(true);
+    await expect(page.isFocused(wrapper.findTabLinkByIndex(1).toSelector())).resolves.toBe(false);
+  })
+);
+
+test(
+  'verifies focus moves from action to tab content on tab press',
+  setupTest(async page => {
+    await page.focusTabHeader();
+    await page.click(wrapper.findTabLinkByIndex(6).toSelector());
+    await page.keys(['ArrowRight']);
+    await page.keys(['ArrowRight']);
+    await page.keys(['Tab']);
+    await expect(page.isFocused(wrapper.findTabContent().toSelector())).resolves.toBe(true);
+  })
+);
+
+test(
+  'verifies dismissible event fires',
+  setupTest(async page => {
+    await page.click(dismissibleWrapper.findTabLinkByIndex(1).toSelector());
+    await page.keys(['ArrowRight']);
+    await page.keys(['Enter']);
+    await expect(page.isFocused(dismissibleWrapper.findTabLinkByIndex(1).toSelector())).resolves.toBe(true);
+  })
+);
+
+test(
+  'prevents user from calling dismiss event twice in tab list of two',
```

Comment:
> What do we test here exactly?

We are testing the component does not invoke the onDismiss event if the user's tab component has a length of <= 1. 

> is that our component's limitation or is it the test page?

Component limitation. I added this because I was re-reading the tech-design and saw this: https://quip-amazon.com/jZhtASlAysv3#QYU9DAKyquI

I think there should be a stopgap measure on the Cloudscape measure to prevent this from happening but the user should be handling this case as well

Reply:
It was discussed initially to enforce a component limitation on removing the single tab. The user would own the actual removal process so we're preventing the onDismiss callback